PRASFiles.jl/src/Systems/read.jl

```diff
@@ -273,29 +275,74 @@ function _systemmodel_core(f::File)
             storages, region_stor_idxs,
             generatorstorages, region_genstor_idxs,
             lines, interface_line_idxs,
-            timestamps)
+            timestamps),type_params
 end
 
 """
 Read a SystemModel from a PRAS file in version 0.5.x - 0.7.x format.
 """
 function systemmodel_0_5(f::File)
 
-    return SystemModel(_systemmodel_core(f)...)
+    systemmodel_0_5_objs, _ = _systemmodel_core(f)
+
+    return SystemModel(systemmodel_0_5_objs...)
 
 end
 
 """
 Read a SystemModel from a PRAS file in version 0.8.x format.
 """
 function systemmodel_0_8(f::File)
+
+    has_demandresponses = haskey(f, "demandresponses")
```

Comment:
I would just check this directly in the if block below, to make it clear why you’re looking for this